### DISCUSSION TITLE
GCP self deployment

### DISCUSSION THREAD

**Discussion by mike12345567** [Budibase/budibase#988] [Ideas]
<p align="center">
<img src="https://user-images.githubusercontent.com/4407001/104211442-fd9a3200-542b-11eb-956a-b831c25c612c.png"  width="128">
</p>

Budibase now supports self hosting with the help of [Docker](https://www.docker.com/)/[Docker compose](https://docs.docker.com/compose/) but we want to keep making it easier to build your ideas with our tools. As part of this we plan to introduce more self hosting deployment options, one option for this is support for GCP (Google Cloud Platform).

If you would be interested in a simple GCP deployment of Budibase this is the place to talk about it! We would love to know your use case for this and why you would prefer this over any other deployment option.

**Comment by Zu-don** [comment]
I'd like more information on how to implement on GCP.

**Reply by shogunpurple** [reply]
Hi @Zu-don - in order to run budibase on GCP, you can bring up a VM in GCP with compose support using this guide:
https://cloud.google.com/community/tutorials/docker-compose-on-container-optimized-os

Once, your VM is up, SSH into it and you can set budibase up as normal:
https://docs.budibase.com/self-hosting/hosting-methods/docker-compose

We are currently working on a kubernetes deployment strategy, which will be available as helm charts over the next few weeks.

**Reply by Zu-don** [reply]
Thanks a lot @shogunpurple !!

**Comment by mdovn** [comment]
Hi, can I host budibase on cloudrun?

**Reply by mjashanks** [reply]
We don't specifically have deployment templates for Cloudrun. There's no reason why it can't be hosted here, it's just that we have never tried it!

If you have experience with it, it would be amazing if you could try that out and let us know.

**Comment by dfanaei** [comment]
Hi, Are you guys able to create a deployment in google cloud marketplace?

**Reply by mjashanks** [reply]
We would love to :) but it's not a priority at this moment. AFAIK there's quite a bit of effort in it.
